src/app/exams/exams-view.component.ts

```diff
@@ -118,6 +121,9 @@ export class ExamsViewComponent implements OnInit, OnDestroy {
       return;
     }
     if (this.isDialog) {
+      if (nextQuestion > this.maxQuestions - 1) {
```

Comment:
With the `return` above on line 121, this conditional will actually always be true so you can remove it.  I think I suggested this, but missed that with the return in the if block above the exam will always be complete.